docs/proposals/object-search.md

```diff
@@ -0,0 +1,160 @@
+### Goals
+* Support single/multi-object search
+* Support across-camera object search
+* Support parallel ReID
+* Support historical video data object search
+* Support multi-camera data joint analysis and decision making
+* Provide a user entry for submitting search tasks and result queries
+
+
+## Proposal
+We propose using Kubernetes Custom Resource Definitions (CRDs) to describe the object search service specification/status and a controller to synchronize these updates between edge and cloud.
+![](./images/object-search-service-crd.png)
+
+### Use Cases
+
+* User can create typical multi-edge collaborative object search applications with providing AI models.
+
+## Design Details
+### CRD API Group and Version
+The `ObjectSearchService` CRD will be namespace-scoped.
+The tables below summarize the group, kind and API version details for the CRD.
+
+* ObjectSearchService
+
+| Field                 | Description             |
+|-----------------------|-------------------------|
+|Group                  | sedna.io     |
+|APIVersion             | v1alpha1                |
+|Kind                   | ObjectSearchService   |
+
+### Object search service CRD
+Below is the CustomResourceDefinition yaml for `ObjectSearchService`:
+
+[crd source](/build/crd-samples/sedna/objectsearchservice_v1alpha1.yaml)
+
+### Object search service type definition
+
+[go source](/pkg/apis/sedna/v1alpha1/objectsearchservice_types.go)
+
+#### Validation
+[Open API v3 Schema based validation](https://kubernetes.io/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definitions/#validation) can be used to guard against bad requests.
+Invalid values for fields (example string value for a boolean field etc) can be validated using this.
+
+Here is a list of validations we need to support :
+1. The `model` specified in the crd should exist in k8s.
```

Comment:
`model` doesn't exist in your provided CR samples.

Reply:
fix